app/kuma-dp/pkg/dataplane/metrics/metrics_format_mapper.go

```diff
@@ -49,24 +50,26 @@ func FromPrometheusMetrics(appMetrics map[string]*io_prometheus_client.MetricFam
 	return openTelemetryMetrics
 }
 
-func gaugeDataPoints(prometheusData []*io_prometheus_client.Metric, extraAttributes []attribute.KeyValue) []metricdata.DataPoint[float64] {
+func gaugeDataPoints(prometheusData []*io_prometheus_client.Metric, extraAttributes []attribute.KeyValue, requestTime time.Time) []metricdata.DataPoint[float64] {
 	var dataPoints []metricdata.DataPoint[float64]
 	for _, metric := range prometheusData {
 		attributes := createOpenTelemetryAttributes(metric.Label, extraAttributes)
 		dataPoints = append(dataPoints, metricdata.DataPoint[float64]{
 			Attributes: attributes,
+			Time:       getTimeOrFallback(metric.TimestampMs, requestTime),
 			Value:      metric.Gauge.GetValue(),
 		})
 	}
 	return dataPoints
 }
 
-func summaryDataPoints(prometheusData []*io_prometheus_client.Metric, extraAttributes []attribute.KeyValue) []metricdata.SummaryDataPoint {
+func summaryDataPoints(prometheusData []*io_prometheus_client.Metric, extraAttributes []attribute.KeyValue, requestTime time.Time) []metricdata.SummaryDataPoint {
 	var dataPoints []metricdata.SummaryDataPoint
 	for _, metric := range prometheusData {
 		attributes := createOpenTelemetryAttributes(metric.Label, extraAttributes)
 		dataPoints = append(dataPoints, metricdata.SummaryDataPoint{
 			Attributes:     attributes,
+			Time:           getTimeOrFallback(metric.TimestampMs, requestTime),
```

Comment:
Question, don't we need to set `StartTime` aswell?

Reply:
https://opentelemetry.io/docs/specs/otel/metrics/data-model/#gauge as otel docs mention this is optional 
![image](https://github.com/user-attachments/assets/635b89f7-666d-4f09-81d8-3a30cc1cd0a4)